src/roi/Roi.ts

```diff
@@ -608,3 +641,105 @@ function getBorders(roi: Roi): { ids: number[]; lengths: number[] } {
     lengths: borderLengths,
   };
 }
+
+function getEllipse(
+  roi: Roi,
+  options: { nbSD: number; scale: number },
+): Ellipse {
+  let xCenter = roi.centroid.column;
+  let yCenter = roi.centroid.row;
+  let newPoints: number[][] = roi.points;
+
+  let xCentered = newPoints.map((item: number[]) => item[0] - xCenter);
+  let yCentered = newPoints.map((item: number[]) => item[1] - yCenter);
+
+  let centeredXVariance = variance(xCentered, { unbiased: false });
+  let centeredYVariance = variance(yCentered, { unbiased: false });
+
+  let centeredCovariance = covariance(
+    {
+      x: xCentered,
+      y: yCentered,
+    },
+    { unbiased: false },
+  );
+
+  //spectral decomposition of the sample covariance matrix
+  let sampleCovarianceMatrix = [
+    [centeredXVariance, centeredCovariance],
+    [centeredCovariance, centeredYVariance],
+  ];
+  let e = new EigenvalueDecomposition(sampleCovarianceMatrix);
+  let eigenvalues = e.realEigenvalues;
+  let vectors = e.eigenvectorMatrix;
+
+  let rMajor: number;
+  let rMinor: number;
+  let vectorMajor: number[];
+  let vectorMinor: number[];
+  let ellipseSurface: number;
+  if (eigenvalues[0] > eigenvalues[1]) {
+    rMajor = Math.sqrt(eigenvalues[0] * options.nbSD);
+    rMinor = Math.sqrt(eigenvalues[1] * options.nbSD);
+    vectorMajor = vectors.getColumn(0);
+    vectorMinor = vectors.getColumn(1);
+  } else if (eigenvalues[0] < eigenvalues[1]) {
+    rMajor = Math.sqrt(eigenvalues[1] * options.nbSD);
+    rMinor = Math.sqrt(eigenvalues[0] * options.nbSD);
+    vectorMajor = vectors.getColumn(1);
+    vectorMinor = vectors.getColumn(0);
+  } else {
+    // order here does not matter
+    rMajor = Math.sqrt(eigenvalues[1] * options.nbSD);
+    rMinor = Math.sqrt(eigenvalues[0] * options.nbSD);
+    vectorMajor = vectors.getColumn(1);
+    vectorMinor = vectors.getColumn(0);
+  }
+  rMajor *= options.scale;
+  rMinor *= options.scale;
+
+  let majorAxisPoint1 = {
+    x: xCenter + rMajor * vectorMajor[0],
+    y: yCenter + rMajor * vectorMajor[1],
+  };
+  let majorAxisPoint2 = {
+    x: xCenter - rMajor * vectorMajor[0],
+    y: yCenter - rMajor * vectorMajor[1],
+  };
+  let minorAxisPoint1 = {
+    x: xCenter + rMinor * vectorMinor[0],
+    y: yCenter + rMinor * vectorMinor[1],
+  };
+  let minorAxisPoint2 = {
+    x: xCenter - rMinor * vectorMinor[0],
+    y: yCenter - rMinor * vectorMinor[1],
+  };
+
+  ellipseSurface =
+    Math.sqrt(
+      Math.pow(majorAxisPoint1.x - xCenter, 2) +
+        Math.pow(majorAxisPoint1.y - yCenter, 2),
+    ) *
+    Math.sqrt(
+      Math.pow(minorAxisPoint1.x - xCenter, 2) +
+        Math.pow(minorAxisPoint1.y - yCenter, 2),
+    ) *
+    Math.PI;
+  return {
+    rMajor,
```

Comment:
We don't need rMajor and rMinor we will have both length in the minorAxis and majorAxis properties